src/compositor/shells/xdg_v6.rs

```diff
@@ -60,8 +60,23 @@ impl XdgV6ShellManagerHandler for XdgV6ShellManager {
         if is_toplevel {
             with_handles!([(compositor: {compositor})] => {
                 let server: &mut Server = compositor.into();
-                server.views
-                    .push(View::new(Shell::XdgV6(shell_surface.into())));
+                let Server { ref mut seat,
+                             ref mut views,
+                             .. } = *server;
+                views.insert(0, View::new(Shell::XdgV6(shell_surface.into())));
```

Comment:
If this is the data structure we want to go with, I think having a linked list might be more performant (or something that doesn't require shifting all the of the elements over everytime focus changes or when a new view is added).

Though again I think this approach could use a lot of work so don't waste time switching to that unless we can find a better solution.

Reply:
You could push it to the back and then render the list forward and find the surface for input backward.

Reply:
Focusing on another view still requires reordering everything (in the worst case) which is inefficient in a Vector.